src/playgroundrl_envs/games/petting_zoo.py

```diff
@@ -46,26 +47,34 @@ def __init__(
         )  # TODO: don't do this unless we need it
         self.env.reset()
         self.state = self.env.last()
+        self.latest_json_action = {}
 
         for i, agent in enumerate(self.env.agents):
             players[i].agent_name = agent
 
         # for now, just assign the player moving sequentially
-        # TODO: check if this generalizes for all games
         self.player_moving_index = 0
         self.player_ids = list(self.players.keys())
 
     def submit_action(self, raw_action, player_sid=""):
         curr_player = self.players[self.player_ids[self.player_moving_index]]
-        if curr_player.sid != player_sid:
-            # Assert the socket has the right to make actions for this player
-            raise PlaygroundInvalidActionException("Not player's turn")
-
-        # parse the action
-        action = pickle.loads(codecs.decode(raw_action.encode(), "base64"))
-        print(type(action))
-
-        self.env.step(action)  # TODO: check if this was performed smoothly
+        # if curr_player.sid != player_sid:
+        #     # Assert the socket has the right to make actions for this player
+        #     raise PlaygroundInvalidActionException("Not player's turn")
+                       
+        player_id = self.player_ids[self.player_moving_index]
+        pettingzoo_agent = self.players[player_id].agent_name
+        action_space = self.env.action_space(agent = pettingzoo_agent)
+
+        if self.self_training:
+            action = pickle.loads(codecs.decode(raw_action.encode(), "base64"))  
+        else:
+            action = json.loads(raw_action)    
```

Comment:
i havent tested this, but i think there might be a case not captured in this. what if a model is playing but not via self_training? we would still want to use the unpickling on their actions submitted. it seems like we want to use json.loads only when the action is submitted by a human player on the frontend.